Classes/Controllers/PBGitHistoryController.m

```diff
@@ -628,6 +632,42 @@ - (NSArray *)menuItemsForPaths:(NSArray *)paths
 	return menuItems;
 }
 
+- (void)didDoubleClickCommit:(id)sender
+{
+	id refish = nil;
+	
+	// If they double-clicked the row with ref labels in we need to check if they clicked a ref label.
+	if ([commitList.tableColumns[commitList.clickedColumn].identifier isEqualToString:@"SubjectColumn"])
+	{
+		// Get the cell.
+		PBGitRevisionCell *cell = (PBGitRevisionCell*)[commitList preparedCellAtColumn:commitList.clickedColumn row:commitList.clickedRow];
+
+		// Get the click position relative to the cell.
+		NSEvent* event = [NSApp currentEvent];
+		NSPoint windowClickLoc = event.locationInWindow;
+		NSPoint tableClickLoc = [commitList convertPoint:windowClickLoc fromView:nil];
+		
+		CGFloat columnX = [commitList rectOfColumn:commitList.clickedColumn].origin.x;
+		CGFloat cellX = tableClickLoc.x - columnX;
+		
+		// IF they clicked a ref, get it.
+		int refIdx = [cell indexAtX:cellX];
+		if (refIdx >= 0)
+			refish = [cell.objectValue.refs objectAtIndex:refIdx];
+	}
+	
+	// If they didn't click a ref, get the commit for that row.
+	if (refish == nil)
+		refish = [commitController.arrangedObjects objectAtIndex:commitList.clickedRow];
+	
+	// Check out the commit or ref.
+	NSError *error = nil;
+	BOOL success = [repository checkoutRefish:refish error:&error];
```

Comment:
`refish` might still be `nil` here, which would be asserting (as in "programmer error"). It's not that apparent, and I'm not actually sure it can happen but if it does, your process is done 😉.

I think I would reorder the code thusly :
- initialize the refish from the clickedRow
- maybe get the refish from the row label
- return if there's still no refish
- perform checkout